pkg/controller/cluster/agent/ports.go

```diff
@@ -0,0 +1,253 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	kubeletPortRangeConfigMapName = "k3k-kubelet-port-range"
+	webhookPortRangeConfigMapName = "k3k-webhook-port-range"
+)
+
+type PortAllocator struct {
+	KubeletCM        *v1.ConfigMap
+	WebhookCM        *v1.ConfigMap
```

Comment:
What about adding directly the Client in the struct? It's needed for its operations, and I think is ok to embed it.
```suggestion
type PortAllocator struct {
	ctrlruntimeclient.Client
	KubeletCM	*v1.ConfigMap
	WebhookCM	*v1.ConfigMap
```